Content.Server/LandMines/LandMineSystem.cs

```diff
@@ -1,43 +1,75 @@
-using Content.Shared.StepTrigger;
 using Content.Shared.StepTrigger.Systems;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.Map;
 using Robust.Shared.Player;
 
 namespace Content.Server.LandMines;
 
 public sealed class LandMineSystem : EntitySystem
 {
+    [Dependency] private readonly IEntityManager _entityManager = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly TransformSystem _transformSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
     [Dependency] private readonly SharedPopupSystem _popupSystem = default!;
     [Dependency] private readonly TriggerSystem _trigger = default!;
 
-
     public override void Initialize()
     {
-        SubscribeLocalEvent<LandMineComponent, StepTriggeredEvent>(HandleTriggered);
-        SubscribeLocalEvent<LandMineComponent, StepTriggerAttemptEvent>(HandleTriggerAttempt);
+        SubscribeLocalEvent<LandMineComponent, StepOffTriggeredEvent>(HandleStepOffTriggered);
+        SubscribeLocalEvent<LandMineComponent, StepTriggeredEvent>(HandleStepTriggered);
+
+        SubscribeLocalEvent<LandMineComponent, StepTriggerAttemptEvent>(HandleStepTriggerAttempt);
     }
 
-    private static void HandleTriggerAttempt(
-        EntityUid uid,
-        LandMineComponent component,
-        ref StepTriggerAttemptEvent args)
+    private void HandleStepOffTriggered(EntityUid uid, LandMineComponent component, ref StepOffTriggeredEvent args)
     {
-        args.Continue = true;
+        if (component.ExplodeImmediately)
+        {
+            return;
+        }
+
+        _trigger.Trigger(uid, args.Tripper);
     }
 
-    private void HandleTriggered(EntityUid uid, LandMineComponent component, ref StepTriggeredEvent args)
+    private void HandleStepTriggered(EntityUid uid, LandMineComponent component, ref StepTriggeredEvent args)
     {
-        // This doesn't use TriggerOnStepTrigger since we don't want to display the popup if nothing happens
-        // and I didn't feel like making an `AfterTrigger` event
-        if (_trigger.Trigger(uid, args.Tripper))
+        ShowLandminePopup(uid, args.Tripper);
+        PlayLandmineActivatedSound(uid, component);
+
+        if (!component.ExplodeImmediately)
         {
-            _popupSystem.PopupCoordinates(
+            return;
+        }
+
+        _trigger.Trigger(uid, args.Tripper);
+    }
+
+    private void ShowLandminePopup(EntityUid uid, EntityUid tripper)
+    {
+        _popupSystem.PopupCoordinates(
                 Loc.GetString("land-mine-triggered", ("mine", uid)),
                 Transform(uid).Coordinates,
-                args.Tripper,
+                tripper,
                 PopupType.LargeCaution);
-        }
     }
-}
 
+    private void PlayLandmineActivatedSound(EntityUid uid, LandMineComponent component)
+    {
+        var xform = _entityManager.GetComponent<TransformComponent>(uid);
```

Comment:
use proxy Transform + all this should be useless if the sound still gets played after mine is deleted, 1 line to play sound on entity uid. if not then play at coords

Reply:
For now, I saw there's a function to PlayPvs() and used that; will that work as intended with maxDistance set in audioparams?

Also, it's pretty neat that sound ends as soon as landmine is destroyed, that's a logical thing to happen 👍🏼